pkg/templatelookup/moduletemplateinfolookup/by_version_test.go

```diff
@@ -0,0 +1,106 @@
+package moduletemplateinfolookup_test
+
```

Comment:
Also here, some branches deserve to be covered:
* When filterTemplatesByVersion returns an error due to failing to list module templates
* When filterTemplatesByVersion returns an error due to more than one template candidate
* When filterTemplatesByVersion returns an error due to no templates found
* When filterTemplatesByVersion returns an error due to the template being marked as mandatory

Reply:
https://github.com/kyma-project/lifecycle-manager/issues/2202